include/seqan3/alignment/pairwise/align_result.hpp

```diff
@@ -34,210 +34,154 @@
-                  "for the id and the alignment score");
-
-    //!\brief The base tuple type.
-    using base_type = detail::transfer_template_args_onto_t<output_type_list_t, std::tuple>;
-
-    //!\brief Inheriting the constructors of the base type.
-    using base_type::base_type;
+private:
+    //! \brief The alignment identifier.
+    id_t id;
+    //! \brief The alignment score.
+    score_t score;
+    //! \brief The end coordinate of the alignment.
+    std::optional<coordinate_t> end_coordinate;
+    //! \brief The begin coordinate of the alignment.
+    std::optional<coordinate_t> begin_coordinate;
+    //! \brief The alignment trace, i.e. the actual base pair matching.
+    std::optional<trace_t> trace;
+
+public:
+    /*!\brief Constructor with all available fields.
+     * \param id_               The alignment identifier.
+     * \param score_            The alignment score.
+     * \param end_coordinate_   The end coordinate of the alignment.
+     * \param begin_coordinate_ The begin coordinate of the alignment.
+     * \param trace_            The alignment trace, i.e. the actual base pair matching.
+     */
+    align_result(id_t const id_,
+                 score_t const score_,
+                 coordinate_t const end_coordinate_,
+                 coordinate_t const begin_coordinate_,
+                 trace_t const & trace_)
+        : id{id_}, score{score_}, end_coordinate{end_coordinate_}, begin_coordinate{begin_coordinate_}, trace{trace_}
+    {}
+
+    /*!\brief Constructor without trace.
+     * \param id_               The alignment identifier.
+     * \param score_            The alignment score.
+     * \param end_coordinate_   The end coordinate of the alignment.
+     * \param begin_coordinate_ The begin coordinate of the alignment.
+     */
+    align_result(id_t const id_,
+                 score_t const score_,
+                 coordinate_t const end_coordinate_,
+                 coordinate_t const begin_coordinate_)
+        : id{id_}, score{score_}, end_coordinate{end_coordinate_}, begin_coordinate{begin_coordinate_}, trace{}
+    {}
+
+    /*!\brief Constructor without trace and begin coordinate information.
+     * \param id_               The alignment identifier.
+     * \param score_            The alignment score.
+     * \param end_coordinate_   The end coordinate of the alignment.
+     */
+    align_result(id_t const id_,
+                 score_t const score_,
+                 coordinate_t const end_coordinate_)
+        : id{id_}, score{score_}, end_coordinate{end_coordinate_}, begin_coordinate{}, trace{}
+    {}
+
+    /*!\brief Constructor with identifier and score only.
+     * \param id_               The alignment identifier.
+     * \param score_            The alignment score.
+     */
+    align_result(id_t const id_,
+                 score_t const score_)
+        : id{id_}, score{score_}, end_coordinate{}, begin_coordinate{}, trace{}
+    {}
+
```

Comment:
Please add a default constructor. 
In this case the access would always throw and id would be -1. -> type deduced.